### PR TITLE
Adding tensorboard tracking/visualization and nnx display example

### PR DESCRIPTION
**Pull request by dillonroach**
Starting from basic model/data in getting_started_with_jax_for_AI and modifying as appropriate to showcase tensorboard integration/tracking as well as basic nnx.display examples.

### CODE REVIEW

File: docs/JAX_visualizing_models_metrics.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# JAX and Tensorboard / NNX Display
+
++++
+
+To keep things straightforward and familiar, we reuse the model and data from the 'getting_started_with_jax_for_AI' notebook - if you haven't read that yet and want the primer, start there before returning.
```
Comment (vfdev-5) [inline docs/JAX_visualizing_models_metrics.md:19]: Maybe, add a link to "getting_started_with_jax_for_AI"

File: docs/JAX_visualizing_models_metrics.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.15.2
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# JAX and Tensorboard / NNX Display
```
Comment (vfdev-5) [inline docs/JAX_visualizing_models_metrics.md:15]: In the beginning or in the end it could be good to have some tb API links to read.

**Review by vfdev-5 — COMMENTED** [review]
So, you do not want to showcase tbX in addition to pure tb/tf ?

File: docs/JAX_visualizing_models_metrics.md (hunk 1 above)
Comment (dillonroach) [reply docs/JAX_visualizing_models_metrics.md:19]: Updated and included.

File: docs/JAX_visualizing_models_metrics.md (hunk 2 above)
Comment (dillonroach) [reply docs/JAX_visualizing_models_metrics.md:15]: Updated and included - at the end we have official TB getting-started and the readthedocs for Treescope

File: docs/JAX_visualizing_models_metrics.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
++++
+
+To keep things straightforward and familiar, we reuse the model and data from the 'getting_started_with_jax_for_AI' notebook - if you haven't read that yet and want the primer, start there before returning.
```
Comment (jakevdp) [inline docs/JAX_visualizing_models_metrics.md:19]: We should link to https://jax-ai-stack.readthedocs.io/en/latest/getting_started_with_jax_for_AI.html and remove the github source tree links below.

File: docs/JAX_visualizing_models_metrics.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
++++
+
+## Extra Resources
+
+For further information about `TensorBoard` see [https://www.tensorflow.org/tensorboard/get_started](https://www.tensorflow.org/tensorboard/get_started)
+
+For more about `nnx.display()`, which calls Treescope under the hood, see [https://treescope.readthedocs.io/en/stable/](https://treescope.readthedocs.io/en/stable/)
+
+```{code-cell} ipython3
```
Comment (jakevdp) [inline docs/JAX_visualizing_models_metrics.md:250]: Remove the empty code cell

**Review by jakevdp — COMMENTED** [review]
Looks great! Just a couple comments below.

File: docs/JAX_visualizing_models_metrics.md (hunk 3 above)
Comment (dillonroach) [reply docs/JAX_visualizing_models_metrics.md]: Thanks for the catch - updated!

**Comment by jakevdp** [comment]
Looks good! We just need to resolve conflicts and then we can merge this!

**Comment by dillonroach** [comment]
@jakevdp conflicts resolved

**Review by jakevdp — APPROVED** [review]
Thanks!

**Pull request merged**